Source/Foundation/bsfUtility/Utility/BsUtil.cpp

```diff
@@ -8,7 +8,7 @@ namespace bs
 	String md5(const WString& source)
 	{
 		MD5 md5;
-		md5.update((UINT8*)source.c_str(), (UINT32)source.length() * sizeof(WString::value_type));
+		md5.update((UINT8*)source.data(), (UINT32)source.length() * sizeof(WString::value_type));
```

Comment:
It's theoretically allowed for c_str() to involve an allocation and copy. Since md5.update() is being given the length as well as the data, just calling data() is perfectly safe, and avoids the possibility of an STL implementation allocating under the hood.